File: csharp/sieve/Sieve.cs
using System.Collections.Generic;
using System.Linq;

public class Sieve
{
	public static int[] Primes(int limit)
	{
		List<int> range = Enumerable.Range(2, limit - 1).ToList();
		for (int i = 2; i < range.Count; i++)
		{
			for (int j = i; j < range.Count; j++)
			{
				range.Remove(i * j);
			}
		}
		return range.ToArray();
	}
}

File: ruby/sieve/sieve.rb
class Sieve
  attr_reader :primes

  def initialize(limit)
    @primes = (2..limit).to_a
    @primes.each do |x|
      (x..limit).each do |y|
        @primes.delete(x * y)
      end
    end
    @primes
  end
end

module BookKeeping
  VERSION = 1
end
